mcsema/BC/Function.cpp

```diff
@@ -260,6 +288,132 @@ static llvm::BasicBlock *GetOrCreateBlock(TranslationContext &ctx,
   return block;
 }
 
+static void LiftExceptionFrameLP(TranslationContext &ctx,
+                                 const NativeFunction *cfg_func) {
+
+  if (cfg_func->eh_frame.size() > 0) {
+
+    // The personality function `__gxx_personality_v0` is lifted as global variable. Erase the
+    // variable before declaring it as the function.
+    if (auto llvm_used = gModule->getGlobalVariable("__gxx_personality_v0")) {
+      llvm_used->eraseFromParent();
+    }
+
+    auto personality = gModule->getFunction("__gxx_personality_v0");
+
+    if (personality == nullptr) {
+      personality = llvm::Function::Create(
+          llvm::FunctionType::get(llvm::Type::getInt32Ty(*gContext), true),
```

Comment:
What is the personality function, and does it always return an int32?

Reply:
The personality function is responsible for decoding the exception info  (type, action, landing pad etc) from the exception table. It returns enum `_Unwind_Reason_Code`. I think int32 should work. The personality function will be different on the different platform (such as Windows). Maybe its good idea to have the wrapper to get the correct name and types.